frontend/Rhyolite/Frontend/Form.hs

```diff
@@ -153,39 +157,49 @@ validateUniqueName name otherNames = fromEither $ do
   guardEither () $ not $ Set.member name otherNames
   return name
 
-data ValidationConfig t m e a = ValidationConfig
+-- | Configure how to perform validation of an input widget
+--
+-- - `e` is the error type of the validation
+-- - `a` is the result type of the validation
+-- - `v` is the value used (typically `Text`) internally by the widget
+data ValidationConfig t m e a v = ValidationConfig
   { _validationConfig_feedback :: Either (Dynamic t e) (Dynamic t a) -> m ()
   -- ^ For displaying the error in the browser with manual styling.
   , _validationConfig_errorText :: e -> Text
   -- ^ For the base HTML form validation, in which errors are non-empty strings.
-  , _validationConfig_validation :: Dynamic t Text -> DynValidation t e a
+  , _validationConfig_validation :: Dynamic t v -> DynValidation t e a
   -- ^ Input is always being reevaluated, including when external dynamics
   -- "mixed in" with this change. But rather than pushing changes downstream,
   -- downstream needed to ask for them (poll) with the 'validate' field.
-  , _validationConfig_validationM :: Maybe (Dynamic t Text -> m (DynValidation t e a))
+  , _validationConfig_validationM :: Maybe (Dynamic t v -> m (DynValidation t e a))
   -- ^ This validation allows for the use of monadic effects (e.g. ask a
   -- server). The results of `_validationConfig_validatation` and
   -- `_validationConfig_validationM` will be combined by `*>`.
   , _validationConfig_initialAttributes :: Map AttributeName Text
   , _validationConfig_validAttributes :: Map AttributeName Text
   , _validationConfig_invalidAttributes :: Map AttributeName Text
-  , _validationConfig_initialValue :: Text
-  , _validationConfig_setValue :: Maybe (Event t Text)
+  , _validationConfig_initialValue :: v
+  , _validationConfig_setValue :: Maybe (Event t v)
   , _validationConfig_validate :: Event t ()
   -- ^ When to show validations and open the gate so downstream gets a new
   -- result. Fresh errors is the price for fresh results.
   }
 
-defValidationConfig :: DomBuilder t m => ValidationConfig t m Text a
-defValidationConfig = ValidationConfig
+-- | Create a `ValidationConfig` with whatever defaults possible
+defValidationConfig
+  :: DomBuilder t m
+  => v
+  -- ^ Initial value to use in the widget
+  -> ValidationConfig t m Text a v
+defValidationConfig ini = ValidationConfig
   { _validationConfig_feedback = const blank
   , _validationConfig_errorText = id
   , _validationConfig_validation = const $ toDynValidation $ pure $ Left "Validation not configured"
   , _validationConfig_validationM = Nothing
   , _validationConfig_initialAttributes = mempty
   , _validationConfig_validAttributes = mempty
   , _validationConfig_invalidAttributes = mempty
-  , _validationConfig_initialValue = ""
```

Comment:
Maybe require Monoid v here and add a second `mkBaseValidationConfig` or something without that constraint that takes an argument.

Reply:
Okay. Does it make sense to deprecate `defValidationConfig` in the process? Because I wonder about the value of having two functions (with non-similar names to boot).

Reply:
`mkValidationConfig` seems like a better name since "base" is not in the type's name. You can't deprecate the argument-taking one because `v` may not be monoidal.

Reply:
> You can't deprecate the argument-taking one

Yea, but I meant: deprecate the _other_ one (one that has the Monoid constraint).

Reply:
Isn't that the one that @eskimor is proposing you *add*? Maybe I missed something.

Reply:
I think not having to provide an argument for the common case is useful and would make this change less breaking (maybe not breaking at all). So I don't think we need to deprecate anything.